src/ORM/Table.php

```diff
@@ -1219,21 +1220,54 @@ public function belongsToMany(string $associated, array $options = []): BelongsT
      * ### Calling finders
      *
      * The find() method is the entry point for custom finder methods.
-     * You can invoke a finder by specifying the type:
+     * You can invoke a finder by specifying the type.
+     *
+     * This will invoke the `findPublished` method:
      *
      * ```
      * $query = $articles->find('published');
      * ```
      *
-     * Would invoke the `findPublished` method.
+     * ## Typed finder arguments
+     *
+     * Finders can have typed parameters separate from the `$options` array and the
+     * `$options` parameter can be optional if not needed:
+     *
+     * ```
+     * $query = $articles->find('byCategory', $category);
+     * ```
+     *
+     * Here, the finder "findByCategory" does not have an `$options` parameter:
+     *
+     * ```
+     * function findByCategory(SelectQuery $query, int $category): SelectQuery
+     * {
+     *     return $query;
+     * }
+     * ```
+     *
+     * If you need to pass query options, just add the typed arguments after `$options`:
+     *
+     * ```
+     * $query = $articles->find('byCategory', [...], $category);
+     * ```
+     *
+     * Here, the finder "findByCategory" does have an `$options` parameter:
+     *
+     * ```
+     * function findByCategory(SelectQuery $query, array $options, int $category): SelectQuery
```

Comment:
In the spirit of doing what is best going forward, I would like to propose we **drop** support for $options entirely. Instead we only support named parameters in 5.x. This is a meaningful break in backwards compatibility, but I think it will provide a long term improvement in developer experience. With named parameters we get typed checked optional parameter support. It's everything we ever wanted from `$options`.

I feel like deprecating `$options` in 4.5 is something we can do. Users that would want to upgrade to 5.x will need to upgrade PHP before they can use named parameters. I don't feel bad about this as encouraging them to upgrade PHP will be something they would need to do anyways.

Reply:
Making $options options is definitely complicated. It looks like code we don't want to support.

Reply:
Are you suggesting we drop $options and force users to call applyOptions() directly? I'm a little confused when you say drop it but then say support it with named parameters.

Reply:
I would be okay with dropping $options if we retain a way to still be able to pass values for the query caluses to the find() call. Something like `find('foo', select: $a, order: $b, custom: 'bar')` instead.

Reply:
We could do that. All of the query-specific options would then become reserved keywords for parameter names.